ERCS/erc-7621.md

```diff
@@ -0,0 +1,140 @@
+
+withdrawETH(uint256 _liquidity): Allows BTS LP holders to withdraw ETH from the basket, equivalent to the value of their BTS LP tokens.
+
+rebalance(address[] memory _newTokens, uint256[] memory _newWeights): The owner can manually adjust the types and weights of tokens in the basket.
+
+getTokenDetails: Returns details of a token at a given index (address and weight).
+
+totalTokens: Returns the total number of tokens in the basket.
+
+#### Distribution of BTS tokens
+
+The distribution of BTS tokens occurs during the mint function. The creator sends ETH to the contract, and the contract then swaps the ETH for user designated [ERC-20](./eip-20.md) tokens based on predefined weights. After the swaps, it mints a BTS token (NFT) for the sender using the initialize function of the BTS contract as well as BTS LP tokens. This BTS token is a representation of ownership and allows management of the BTS.
+
+#### Distribution of BTS LP tokens
+
+The distribution of BTS LP tokens occurs during the mint and contribute functions. After swapping ETH for the relevant [ERC-20](./eip-20.md) tokens, the contract mints BTS LP tokens (representing the user's share of the total BTS), and then mints BTS LP tokens using the mint function of the IBTSPair contract. The BTS LP tokens represent the liquidity provided by the user to the specified BTS pair (btsPair). The distribution is logged through the ContributedToBTS event.
+
+#### Burning the BTS LP tokens
+
+The burning of BTS LP tokens occurs during the withdraw function. Users can call this function, specifying the amount of BTS LP tokens they want to withdraw. The contract then transfers the specified amount of BTS LP tokens from the user to the BTS pair contract (btsPair). The burn function of the IBTSPair contract is called, which decreases the user's BTS LP token balance and returns an array of amounts representing the underlying tokens withdrawn. This array of amounts is logged through the WithdrawnFromBTS event.
+
+#### Events
+
+ContributedToBTS(address indexed bts, uint256 amount): Event when someone adds ETH to the basket.
+
+WithdrawnFromBTS(address indexed bts, uint256[] amounts): Event when BTS LP holder withdraws tokens from the basket.
+
+WithdrawnETHFromBTS(address indexed bts, uint256 amount): Event when BTS LP holder withdraws ETH from the basket.
+
+RebalanceBTS(address indexed bts, uint256[] oldWeights, uint256[] newWeights): Event when the owner manually rebalances the basket.
+
+#### Storage
+
+btsPair: A public variable storing the address of the associated BTS pair contract.
+
+autoRebalanceEnabled: A public variable indicating whether auto-rebalancing is enabled or not.
+
+mapping(address => bool) private _isTokenPresent: A private mapping indicating whether a token is present or not.
+
+TokenDetails: A structure holding details about the tokens, including arrays for token addresses and their corresponding weights.
+
+_tokenDetails: An instance of the TokenDetails structure.
+
```

Comment:
```suggestion

```

This looks like implementation details. Contracts should be free to use whatever storage layout they want.